problem-json/src/test/groovy/io/micronaut/problem/ConstraintViolationProblemSpec.groovy

```diff
@@ -0,0 +1,70 @@
+package io.micronaut.problem
+
+import io.micronaut.context.annotation.Requires
+import io.micronaut.core.annotation.Introspected
+import io.micronaut.core.annotation.NonNull
+import io.micronaut.core.annotation.Nullable
+import io.micronaut.core.type.Argument
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.HttpStatus
+import io.micronaut.http.annotation.Body
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Post
+import io.micronaut.http.annotation.Status
+import io.micronaut.http.client.exceptions.HttpClientResponseException
+import org.zalando.problem.DefaultProblem
+
+import javax.inject.Singleton
+import javax.validation.Valid
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.NotNull
+
+class ConstraintViolationProblemSpec extends EmbeddedServerSpecification {
+
+    @Override
+    @Nullable
+    String getSpecName() {
+        'ConstraintViolationProblemSpec'
+    }
+
+    void "ConstraintViolationException is rendered as a problem"() {
+        when:
+        Argument<?> okArg = Argument.of(String)
+        Argument<?> errorArg = Argument.of(Map)
+        client.exchange(HttpRequest.POST('/contact', new ContactCreateForm()), okArg, errorArg)
+
+        then:
+        HttpClientResponseException e = thrown()
+        e.status == HttpStatus.BAD_REQUEST
+        e.response.contentType.isPresent()
+        e.response.contentType.get().toString() == 'application/problem+json'
+
+        when:
+        Optional<Map> bodyOptional = e.response.getBody(errorArg)
+
+        then:
+        bodyOptional.isPresent()
+        bodyOptional.get().keySet().size() == 4
+        bodyOptional.get()['status'] == 400
+        bodyOptional.get()['title'] == 'Constraint Violation'
+        bodyOptional.get()['type'] == "https://zalando.github.io/problem/constraint-violation"
```

Comment:
Can you also assert the content of the `violations` field?

Reply:
Done